### PR TITLE
Feature/day 4

### PR DESCRIPTION
**Pull request by ayushpandey-successive**
Questions on Array and Strings

1. Write a program to remove duplicate charecters from a string ("Hello Yellow" => "Helo Yw")
ques1

2.Write a program to reverse a string ("Hello John" => "olleH nhoJ")
ques2

3. Write a program to give count of variables in a string ("abcabcdabbcc" => "a3b4c4d1")
ques3

4. Write a program to convert given string to upperCase OR lowerCase
ques4

5. Write a program to convert given string in camel case ("hello john doe" => helloJohnDoe)
ques5

6. Write a program to find sum of an array
ques6

7. Write a program to find index of duplicate elements in an array
ques7

8. Write a program to remove dupliacte elements from an array
ques8

9. Write a program to find the last duplicate index in an array
ques9

10. Write a program to concatenate an input of array of arrays
ques10

11. Write a program to sort an array (bubble sort)
ques11

12. Write a program to display intersection of two array
ques12

13. Write a program to return inverse of an array
ques13

14. Write a program to reverse an array
ques14

### CODE REVIEW

File: day-4/ques-1.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//Problem Statement: Write a program to remove duplicate charecters from a string ("Hello Yellow" => "Helo Yw")
+
+const str = "Hello Yellow";
+const str1 = str.split("");
+//console.log(str1 );
```
Comment (saurabh-yadav-successive) [inline day-4/ques-1.js]: remove unnecessary commented code

File: day-4/ques-1.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+let str2 = [];
+for (let i = 0; i < str1.length; i++) {
```
Comment (saurabh-yadav-successive) [inline day-4/ques-1.js]: why we are not using map().

File: day-4/ques-13.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+//1Problem Statement: Write a program to return inverse of an array
+
```
Comment (saurabh-yadav-successive) [inline day-4/ques-13.js]: what is expected result? please comment out

File: day-4/ques-13.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const arr = [3, 4, 2, 0, 1];
+const inverseArr = new Array(arr.length);
+
+const makeInverseArray = (arr) => {
+  for (i in arr) {
+    inverseArr[arr[i]] = i;
+  }
+};
+
+const checkInverse = (arr, inverseArr) => {
```
Comment (saurabh-yadav-successive) [inline day-4/ques-13.js:13]: what is the use of this function

File: day-4/ques-1.js (hunk 1 above)
Comment (ayushpandey-successive) [reply day-4/ques-1.js]: resolved

File: day-4/ques-1.js (hunk 2 above)
Comment (ayushpandey-successive) [reply day-4/ques-1.js]: resolved

File: day-4/ques-13.js (hunk 4 above)
Comment (ayushpandey-successive) [reply day-4/ques-13.js:13]: The question is about how can we inverse an array:
Inverse of an array means that if in an array we swap the value of the array at a particular index with the index of the element, and then we obtain the same array then we can call it as the irreversible array.
eg: [3,4,2,0,1]
in this: when we swap the array elements with the index values in another array:
invArr = [,,,,]
then referring the original array, we place 0 value at index 3 in invArray
invArr = [,,,0,]
Similarly when we refer the index 1 of original array we place 1 at index 4 i.e.
invArr = [,,,0,1],
similarly referring the index 2,3,4 we get the inverse array: invArr = [3,4,2,0,1]

File: day-4/ques-13.js (hunk 3 above)
Comment (ayushpandey-successive) [reply day-4/ques-13.js]: resolved

**Review by saurabh-yadav-successive — APPROVED** [review]

**Pull request merged**